vm-validator/src/vm_validator.rs

```diff
@@ -35,35 +34,23 @@ pub trait TransactionValidation: Send + Sync + Clone {
     fn notify_commit(&mut self);
 }
 
-#[ouroboros::self_referencing(pub_extras)]
-struct CachedStateView {
-    smt: Box<SparseMerkleTree<AccountStateBlob>>,
-    #[borrows(smt)]
-    #[covariant]
-    state_view: VerifiedStateView<'this, DpnProto>,
-}
-
-impl CachedStateView {
-    pub fn create(db_reader: &Arc<dyn DbReader<DpnProto>>) -> Self {
-        let (version, state_root) = db_reader.get_latest_state_root().expect("Should not fail.");
-        let smt = SparseMerkleTree::new(state_root);
-        Self::new(Box::new(smt), move |smt| {
-            VerifiedStateView::new(
-                StateViewId::TransactionValidation {
-                    base_version: version,
-                },
-                Arc::clone(db_reader),
-                Some(version),
-                state_root,
-                smt,
-            )
-        })
-    }
+fn latest_state_view(db_reader: &Arc<dyn DbReader<DpnProto>>) -> VerifiedStateView<DpnProto> {
```

Comment:
remove the ouroboros dependencies as well?

Reply:
oh, looks like someone as good as me has done that already